src/esdtToken.ts

```diff
@@ -36,17 +48,96 @@ export class ESDTToken {
         canPause: boolean,
         canFreeze: boolean,
         canWipe: boolean
-    }) {
-        let esdtToken = new ESDTToken(response);
-        return esdtToken
+    }): ESDTToken {
+        return new ESDTToken({
+            token: response.token,
+            name: response.name,
+            type: TokenType[response.type as keyof typeof TokenType],
+            owner: new Address(response.owner),
+            minted: new BigNumber(response.minted),
+            burnt: new BigNumber(response.burnt),
+            decimals: response.decimals,
+            isPaused: response.isPaused,
+            canUpgrade: response.canUpgrade,
+            canMint: response.canMint,
+            canBurn: response.canBurn,
+            canChangeOwner: response.canChangeOwner,
+            canPause: response.canPause,
+            canFreeze: response.canFreeze,
+            canWipe: response.canWipe,
+        });
     }
 
-    getTokenName() {
+    static fromTokenProperties(tokenIdentifier: string, results: any[]): ESDTToken {
+        let [tokenName, tokenType, owner, totalMinted, totalBurned, ...propertiesBuffers] = results;
+        let properties = parseTokenProperties(propertiesBuffers);
+        return new ESDTToken({
+            token: tokenIdentifier,
+            type: TokenType[tokenType.toString() as keyof typeof TokenType],
+            name: tokenName.toString(),
+            owner,
+            minted: new BigNumber(totalMinted),
+            burnt: new BigNumber(totalBurned),
+            decimals: properties.NumDecimals.toNumber(),
+            isPaused: properties.IsPaused,
+            canUpgrade: properties.CanUpgrade,
+            canMint: properties.CanMint,
+            canBurn: properties.CanBurn,
+            canChangeOwner: properties.CanChangeOwner,
+            canPause: properties.CanPause,
+            canFreeze: properties.CanFreeze,
+            canWipe: properties.CanWipe,
+            canAddSpecialRoles: properties.CanAddSpecialRoles,
+            canTransferNftCreateRole: properties.CanTransferNFTCreateRole,
+            nftCreateStopped: properties.NFTCreateStopped,
+            wiped: properties.NumWiped
+        });
+    }
+
+    getTokenName(): string {
         return this.name;
     }
 
-    getTokenIdentifier() {
+    typeAsString(): string {
+        return TokenType[this.type];
+    }
+
+    getTokenIdentifier(): string {
         return this.token;
     }
 
+    isEgld(): boolean {
+        return this.token == "EGLD";
+    }
+
+    isFungible(): boolean {
+        return !this.isNft();
+    }
+
+    isNft(): boolean {
+        switch (this.type) {
+            case TokenType.FungibleESDT:
+                return false;
+            case TokenType.SemiFungibleESDT:
+            case TokenType.NonFungibleESDT:
+                return true;
+        }
+    }
+}
+
+function parseValue(value: string): any {
+    switch (value) {
```

Comment:
This function is used to check the results of the query to `getTokenProperties`, which, for its properties has values of the form: `NumDecimals-2`, `IsPaused-false`.
Because they come from the ESDT system smart contract, I don't expect them to be malformatted.